pkg/kube/kclient/interfaces.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kclient
+
+import (
+	klabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pkg/kube/controllers"
+)
+
+type Untyped = Informer[controllers.Object]
+
+// Reader wraps a Kubernetes client providing cached read access.
+// This is based on informers, so most of the same caveats to informers apply here.
+type Reader[T controllers.Object] interface {
+	// Get looks up an object by name and namespace. If it does not exist, nil is returned
+	Get(name, namespace string) T
+	// List looks up an object by namespace and labels.
+	// Use metav1.NamespaceAll and klabels.Everything() to select everything.
+	List(namespace string, selector klabels.Selector) []T
+}
+
+type Informer[T controllers.Object] interface {
```

Comment:
It is not clear what's the diff with kubernetes `SharedInformer`

Reply:
If you look at the methods you will see its quite different, the only function in common is AddEventHandler and HasSynced, and both of those functions have different implementations for this.

`Informer` is just the only interface `kclient` implemented, it is a Reader (Get + List) + informer attributes (handler, synced, etc)